src/map/attack.cpp

```diff
@@ -393,16 +393,16 @@ bool CAttack::CheckAnticipated()
         m_victim->StatusEffectContainer->DelStatusEffect(EFFECT_THIRD_EYE);
         return false;
     }
+    auto* weapon         = dynamic_cast<CItemWeapon*>(m_victim->m_Weapons[SLOT_MAIN]);
+    bool  hasValidSeigan = m_victim->StatusEffectContainer->HasStatusEffect(EFFECT_SEIGAN, 0) && weapon && weapon->isTwoHanded();
```

Comment:
This is a little hard to read, I'd probably break this out into:
```cpp
auto* weapon         = dynamic_cast<CItemWeapon*>(m_victim->m_Weapons[SLOT_MAIN]);
bool isValid2HandWeapon = weapon && weapon->isTwoHanded();
bool  hasValidSeigan = isValid2HandWeapon  && m_victim->StatusEffectContainer->HasStatusEffect(EFFECT_SEIGAN, 0);
```

Reply:
Updated.